File: pset6/vigenere/vigenere.py
import cs50
import sys

cypher = ""

# Check if input is correct
if len(sys.argv) > 2:
    print("No spaces between key literals")
    sys.exit(1)

    # make sure that th user has given the Key
if len(sys.argv) == 1:
    print("Please enter a Key")
    sys.exit(1)

while len(sys.argv) == 2:
    key = str(sys.argv[1])
    k = 0

    # makes sure that argument is sequence of letters
    if not key.isalpha():
        print("Please enter as Key only literals")
        sys.exit(1)

    if key:
        # get input string
        text = input("Please input the message to cript: ")

        # iterate over every letter of the string
        for b in text:
            # checks if character is a leter or not
            if k >= len(key):
                k = 0

            shift = key[k]

            if shift.isupper():
                shift = ord(shift)
                shift -= ord('A')

            elif shift.islower():
                shift = ord(shift)
                shift -= ord('a')

            # checks if uppercase
            if b.isupper():
                cypher += chr((ord('A') + (ord(b) - ord('A') + shift) % 26))
                k += 1

            # checks if lowercase
            elif b.islower():
                cypher += chr((ord('a') + (ord(b) - ord('a') + shift) % 26))
                k += 1

            else:
                cypher += b

        # prints the ciphertext
        print("ciphertext: ", cypher)
        sys.exit(0)
        break

    else:
        print("Please enter as Key only literals")
        sys.exit(0)
        break

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{

    //assigning th key variable with the numerical value from th input argument
    string key;
    string text;
    char shift;

    if (argc > 2)
    {
        printf("No spaces between key literals\n");
        return 1;
    }
    if (argc == 1)
    {
        // make sure that th user has given the Key
        printf("Please enter a Key\n");
        return 1;
    }

    while (argc == 2)
    {
        key = argv[1];

        // makes sure that argument is sequence of letters
        for (int i = 0; key[i] != '\0'; i++)
        {

            if (key[i] >= '0' && key[i] <= '9')
            {
                printf("Please enter as Key only literals\n");
                return 1;
            }
        }

        if (key)
        {
            //get input string
            text = get_string("Please input the message to cript: ");

            //iterate over every letter of the string
            for (int j = 0, k = 0; text[j] != '\0'; j++)
            {

                //checks if character is a leter or not
                if (isalpha(text[j]))
                {
                    //prepares the key
                    if (key[k] != '\0')
                    {
                        shift = key[k];
                        if (isupper(shift))
                        {
                            shift -= 'A';
                        }
                        if (islower(shift))
                        {
                            shift -= 'a';
                        }

                        //checks if uppercase
                        if (isupper(text[j]))
                        {
                            text[j] = ('A' + (text[j] - 'A' + shift) % 26);
                            k++;
                        }
                        //checks if lowercase
                        if (islower(text[j]))
                        {
                            text[j] = ('a' + (text[j] - 'a' + shift) % 26);
                            k++;
                        }
                    }
                    else
                    {
                        k = 0;

                        //prepares the key
                        shift = key[k];
                        if (isupper(shift))
                        {
                            shift -= 'A';
                        }
                        if (islower(shift))
                        {
                            shift -= 'a';
                        }

                        //checks if uppercase
                        if (isupper(text[j]))
                        {
                            text[j] = ('A' + (text[j] - 'A' + shift) % 26);
                            k++;
                        }

                        //checks if lowercase
                        if (islower(text[j]))
                        {
                            text[j] = ('a' + (text[j] - 'a' + shift) % 26);
                            k++;
                        }
                    }
                }

                if (ispunct(text[j]))
                {
                }
            }
            //prints the ciphertext
            printf("ciphertext: %s\n", text);
            return 0;
            // break;
        }

        else
        {
            printf("Please enter as Key only literals\n");
            return 1;
            // break;
        }
    }
}